openmc/model/model.py

```diff
@@ -1828,39 +1855,53 @@ def convert_to_multigroup(self, method="material_wise", groups='CASMO-2',
         if isinstance(groups, str):
             groups = openmc.mgxs.EnergyGroups(groups)
 
-        # Make sure all materials have a name, and that the name is a valid HDF5
-        # dataset name
-        for material in self.materials:
-            if material.name is None:
-                material.name = f"material {material.id}"
-            material.name = re.sub(r'[^a-zA-Z0-9]', '_', material.name)
-
-        # If needed, generate the needed MGXS data library file
-        if not Path(mgxs_path).is_file() or overwrite_mgxs_library:
-            if method == "infinite_medium":
-                self._generate_infinite_medium_mgxs(
-                    groups, nparticles, mgxs_path, correction)
-            elif method == "material_wise":
-                self._generate_material_wise_mgxs(
-                    groups, nparticles, mgxs_path, correction)
-            elif method == "stochastic_slab":
-                self._generate_stochastic_slab_mgxs(
-                    groups, nparticles, mgxs_path, correction)
+        # Do all work (including MGXS generation) in a temporary directory
+        # to avoid polluting the working directory with residual XML files
+        with TemporaryDirectory() as tmpdir:
+
+            # Determine if this is a DAGMC geometry. If so, we need to syncronize
+            # the dagmc materials with cells.
+            # TODO: Can this be done without having to init/finalize?
```

Comment:
Not at the moment. Detecting the current material assignments and DAGMC cell IDs as used during transport requires loading the file with the DAGMC library in C++.